cleanlab_studio/utils/data_enrichment/enrich.py

```diff
@@ -33,112 +32,125 @@ def enrich_data(
     You can optionally apply regular expressions to further reformat your metadata beyond raw LLM outputs, or specify that each row of the metadata must be constrained to a particular set of values.
 
     Args:
-        studio: Cleanlab Studio client object, which you must instantiate before calling this method.
-        prompt: Formatted f-string, that contains both the prompt, and names of columns to embed.
+        studio (Studio): Cleanlab Studio client object, which you must instantiate before calling this method.
+        data (pd.DataFrame): A pandas DataFrame containing your data.
+        prompt (str): Formatted f-string, that contains both the prompt, and names of columns to embed.
             **Example:** "Is this a numeric value, answer Yes or No only. Value: {column_name}"
-        regex: A string expression that will be passed into ``re.compile()``, a compiled regular expression, or a list of multiple already compiled regular expressions.
-            Optional expressions passed here will be applied to the raw LLM outputs from your prompt, enabling additional control to better format the final metadata output column.
-            This `regex` argument is useful in settings where you are unable to prompt the LLM to generate valid outputs 100% of the time, but can easily transform the raw LLM outputs to be valid through regular expressions that extract parts of the raw output string.
-            If a list of expressions is provided, the expressions are applied in order and first valid extraction is returned.
-
-            **Note:** Each regex pattern should specify 1 group that represents the desired characters to extract from the raw LLM response using parenthesis like so: ``'(<desired match group pattern>)'``.
-            **Example 1:** ``regex = r'.*The answer is: (Bird|[Rr]abbit).*'`` will extract strings that are the words 'Bird', 'Rabbit' or 'rabbit' after the characters "The answer is: " from the raw LLM response. This might be useful if your prompt instructed the LLM to respond in this format.
-            **Example 2:** ``regex = r'.*(\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b).*'`` will extract an email in the raw LLM response. Similar patterns can be used to extract a specifically structured part of the LLM response.
-        return_values: List of all possible values for the `metadata` column.
+        regex (str | Replacement | List[Replacement], optional): A string, tuple, or list of tuples specifying regular expressions to apply for post-processing the raw LLM outputs.
```

Comment:
I would mention regex is applied and results can be seen in {new_column_name} but the  {new_column_name}log column contains raw outputs pre-regex.